File: 202006-1/cpp/main.cpp
#include <cstdio>
#include <cstring>

using namespace std;

struct Info
{
    int x;
    int y;
    char type[2];
};

struct Line
{
    int a;
    int b;
    int c;
};

Info infoList[1000];
Line lineList[1000];

int num1, num2;

char infoType[2];

int main()
{
    scanf("%d %d", &num1, &num2);

    for (int i = 0; i < num1; i++)
    {
        scanf("%d %d %s", &infoList[i].x, &infoList[i].y, &infoList[i].type[0]);
    }

    for (int i = 0; i < num2; i++)
    {
        scanf("%d %d %d", &lineList[i].a, &lineList[i].b, &lineList[i].c);
    }

    for (int i = 0; i < num2; i++)
    {
        char ok[] = "Yes";

        Line *line = &lineList[i];
        Info *firstInfo = &infoList[0];

        int firstRes = line->a + line->b * firstInfo->x + line->c * firstInfo->y;
        if (firstRes == 0)
        {
            strcpy(ok, "No");
        }
        else
        {
            if ((firstRes > 0 && firstInfo->type[0] == 'A') || (firstRes < 0 && firstInfo->type[0] != 'A'))
            {
                strcpy(infoType, "A");
            }
            else
            {
                strcpy(infoType, "B");
            }

            for (int m = 1; m < num1; m++)
            {
                Info *info = &infoList[m];
                int result = line->a + line->b * info->x + line->c * info->y;
                if (result == 0 || (result > 0 && info->type[0] != infoType[0]) || (result < 0 && info->type[0] == infoType[0]))
                {
                    strcpy(ok, "No");
                    break;
                }
            }
        }
        printf("%s\n", ok);
    }

    return 0;
}

File: 202006-1/java/Main.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

/**
 * @time 2020/8/13 下午12:22
 * @author Limgmk
 * @description
 */
public class Main {
    public static Scanner sc = new Scanner(System.in);

    public static void main(String[] args) {
        List<Info> infoList = new ArrayList<Info>();
        List<Line> lineList = new ArrayList<Line>();

        int num1 = sc.nextInt();
        int num2 = sc.nextInt();

        for(int i = 0; i < num1; i++) {
            Info info = new Info(sc.nextInt(), sc.nextInt(), sc.next());
            infoList.add(info);
        }

        for(int i = 0; i < num2; i++) {
            Line line = new Line(sc.nextInt(), sc.nextInt(), sc.nextInt());
            lineList.add(line);
        }

        for(Line line: lineList) {
            String ok = "Yes";
            Info firstInfo = infoList.get(0);
            int resultFirst = line.a + line.b * firstInfo.x + line.c * firstInfo.y;
            String infoType = "";
            if(resultFirst == 0) {
                ok = "No";
            } else {
                if ((resultFirst > 0 && firstInfo.type.equals("A")) || (resultFirst < 0 && !firstInfo.type.equals("A"))) {
                    infoType = "A";
                } else {
                    infoType = "B";
                }
                for(Info info: infoList) {
                    int result = line.a + line.b * info.x + line.c * info.y;
                    if(result == 0 || (result > 0 && !info.type.equals(infoType)) || (result < 0 && info.type.equals(infoType))) {
                        ok = "No";
                        break;
                    }
                }
            }

            System.out.println(ok);
        }
    }

    static class Info {
        public int x;
        public int y;
        public String type;

        public Info(int x, int y, String type) {
            this.x = x;
            this.y = y;
            this.type = type;
        }
    }

    static class Line {
        public int a;
        public int b;
        public int c;

        public Line(int a, int b, int c) {
            this.a = a;
            this.b = b;
            this.c = c;
        }
    }
}
